services/net/extract-quotes/LLM/LLMService.cs

```diff
@@ -0,0 +1,604 @@
+                    var sentences = Enumerable.Range(0, maxSentenceIndex + 1)
+                                        .Select(i => new Sentence { index = i, entityMentions = new List<EntityMention>() })
+                                        .ToList();
+
+                    // Add entity mentions for speakers to the correct sentence
+                    foreach (var quote in quotes)
+                    {
+                        if (quote.beginSentence >= 0 && quote.beginSentence < sentences.Count && !string.IsNullOrWhiteSpace(quote.canonicalSpeaker))
+                        {
+                            // Avoid adding duplicate speakers to the same sentence if multiple quotes start there
+                            if (!sentences[quote.beginSentence].entityMentions.Any(em => em.text == quote.canonicalSpeaker))
+                            {
+                                sentences[quote.beginSentence].entityMentions.Add(new EntityMention
+                                {
+                                    text = quote.canonicalSpeaker,
+                                    ner = "PERSON" // Assuming speaker is always PERSON
+                                });
+                            }
+                        }
+                    }
+                    this.Logger.LogDebug("Created {count} sentences structure based on quote indices.", sentences.Count);
+                    // --- End Conversion Logic ---
+
+                    var annotationResponse = new AnnotationResponse
+                    {
+                        Quotes = quotes,
+                        Sentences = sentences
+                    };
+
+                    // Log each extracted quote for debugging
+                    foreach (var quote in annotationResponse.Quotes)
+                    {
+                        this.Logger.LogDebug("Extracted quote: Speaker='{speaker}', Text='{text}'",
+                            quote.canonicalSpeaker, quote.text);
+                    }
+
+                    return annotationResponse;
+                }
+                else
+                {
+                    this.Logger.LogWarning("Parsed JSON from model '{Model}' does not contain valid quotes or is empty. JSON: {json}", modelName, jsonString);
+                    return null;
+                }
+            }
+            catch (JsonException ex)
+            {
+                this.Logger.LogError(ex, "Failed to parse LLM response JSON from model '{Model}'. JSON: {json}", modelName, jsonString);
+                return null;
+            }
+        }
+        else
+        {
+            this.Logger.LogWarning("Unable to extract JSON object from LLM response content from model '{Model}'. Content: {content}", modelName, llmResponseContent);
+            return null;
+        }
+    }
+
+    #endregion
+}
+
+/// <summary>
+/// Response from the LLM API
+/// </summary>
+public class LLMResponse
```

Comment:
Create a file for each class.  Don't throw everything into a single file, it makes it more difficult to maintain.